docs/guide/component-testing.md

```diff
@@ -0,0 +1,402 @@
+await expect.element(getByRole('button')).toBeEnabled()
+await expect.element(getByRole('button')).toBeDisabled()
+
+// Text content
+await expect.element(getByRole('heading')).toHaveTextContent('Welcome')
+
+// Attributes
+await expect.element(getByTestId('link')).toHaveAttribute('href', '/home')
+
+// CSS classes and styles
+await expect.element(getByTestId('alert')).toHaveClass('error')
+await expect.element(getByTestId('box')).toHaveStyle({ color: 'red' })
+```
+
+## Testing Library Integration
+
+For frameworks not officially supported by Vitest, you can use Testing Library packages:
+
+```jsx
+// For Solid.js components
+import { render } from '@testing-library/solid'
+import { page } from '@vitest/browser/context'
+
+test('solid component test', async () => {
+  const { baseElement } = render(() => <MyComponent />)
+  const screen = page.elementLocator(baseElement)
+
+  await expect.element(screen.getByText('Hello')).toBeInTheDocument()
+})
+```
+
+## Node.js Component Testing
+
+For faster execution during development, you can test components in Node.js using DOM simulation:
+
+```ts
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+  test: {
+    environment: 'happy-dom', // or 'jsdom'
+  },
+})
+```
+
+::: warning
+Node.js testing with DOM simulation may not catch all browser-specific issues. Use Browser Mode for comprehensive testing.
+:::
+
+## Best Practices
+
+1. **Use Browser Mode for CI/CD** - Ensures tests run in real browser environments
+2. **Test user interactions** - Simulate real user behavior with clicks, typing, etc.
+3. **Test accessibility** - Ensure components are accessible to all users
+4. **Mock external dependencies** - Focus tests on component logic, not external services
+5. **Use meaningful test descriptions** - Make test intent clear for maintainability
+
+## Common Patterns
+
+### Testing Props and Events
+
+```js
+test('emits events on user interaction', async () => {
+  const mockHandler = vi.fn()
+  render(<Button onClick={mockHandler}>Click me</Button>)
+
+  await page.getByRole('button').click()
+
+  expect(mockHandler).toHaveBeenCalledOnce()
+})
+```
+
+### Testing Conditional Rendering
+
+```jsx
+test('shows loading state', async () => {
+  const { getByText, rerender } = render(<DataComponent loading={true} />)
+
+  await expect.element(getByText('Loading...')).toBeInTheDocument()
+
+  rerender(<DataComponent loading={false} data="content" />)
+
+  await expect.element(getByText('content')).toBeInTheDocument()
+})
+```
+
+### Testing Forms
+
+```jsx
+test('validates form input', async () => {
+  render(<ContactForm />)
+
+  const emailInput = page.getByLabelText(/email/i)
+
+  await emailInput.fill('invalid-email')
+  // Trigger validation by focusing away from the input
+  await page.getByRole('button', { name: /submit/i }).click()
+
+  await expect.element(page.getByText('Please enter a valid email')).toBeInTheDocument()
+})
+```
+
+## Debugging Component Tests
```

Comment:
This section suffers the same issues as "Best Practices"

> **Use browser dev tools** - Browser Mode allows full debugging capabilities

A link to devtools? How to use them with browser mode?

> **Inspect rendered output** - Check if components render as expected

How? Is `console.log` enough? Should I look in the UI?

> **Verify selectors** - Ensure element queries target correct elements

What does this mean actually?